src/lib.rs

```diff
@@ -1867,9 +1865,17 @@ impl<T> Clone for InactiveReceiver<T> {
 
 impl<T> Drop for InactiveReceiver<T> {
     fn drop(&mut self) {
-        let mut inner = self.inner.write();
+        let mut inner = write_no_poison(&self.inner);
 
         inner.inactive_receiver_count -= 1;
         inner.close_channel();
     }
 }
+
+fn read_no_poison<T>(lock: &RwLock<T>) -> RwLockReadGuard<'_, T> {
```

Comment:
Thanks for redoing this but I wasn't expecting your to be convinced by my arguments so quickly. :laughing: 


We don't need these methods if we just panic. Just use `inner.read().unwrap()` and `inner.write().unwrap()`.

Reply:
Also since we're changing the behavior about poisoning, it would be good to add a sentence about this in the commit log/PR description.

Sorry for being so pedantic. I'm sure this is the last time you've to re-work before we can merge.

Reply:
> t I wasn't expecting your to be convinced by my arguments so q

Okey, then it make sense to remove `read/write_no_poison` at all =)

Reply:
> > t I wasn't expecting your to be convinced by my arguments so q
> 
> Okey, then it make sense to remove `read/write_no_poison` at all =)

Just to be clear, you mean you will remove these methods as per last suggestion or that you think we should keep the existing behavior of non poisoning? I'll need some arguments for both. :)

The reason I asked you to remove these is because 1. they're just duplicating `unwrap` 2. their names don't make any sense now.

Reply:
> We don't need these methods

Ow, sorry, right, missed begin of your sentence.